### PR TITLE
Ensure DeleteUser is not allowed to Delete Orgs and visa versa

### PR DESCRIPTION
**Pull request by 6543**
as title (extend/come up in #10125)

### CODE REVIEW

**Review by adelowo — APPROVED** [review]
Will be nice to add a test for this though.

**Comment by adelowo** [comment]
Do we want to add the same check to 

https://github.com/go-gitea/gitea/blob/b3c72a7c4a935e805b87ff2e1911429628ace7c5/models/org.go#L257-L275

**Review by guillep2k — COMMENTED** [review]
Will keep the check at routers/api/v1/admin/user.go:231 ?

**Comment by 6543** [comment]
> Will keep the check at routers/api/v1/admin/user.go:231 ?

would say yes -> normal error is 500 and with check it returns 422 in this case ...

File: integrations/prevent_prohibited_deletion_test.go [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+package integrations
+
+import (
+	"code.gitea.io/gitea/models"
+	"github.com/issue9/assert"
+	"testing"
+)
+
+func TestPreventProhibitedDeletion(t *testing.T) {
```
Comment (guillep2k) [inline integrations/prevent_prohibited_deletion_test.go]: This should be a unit test, not integration.

**Comment by 6543** [comment]
@adelowo looks like DeleteOrg had already a check ..
I remove redundancy (deleteOrg is only used in DeleteOrganization) ...
@guillep2k I also moved tests to UnitTests

**Review by lunny — APPROVED** [review]

**Review by guillep2k — APPROVED** [review]

File: models/org.go [hunk 2]
```diff
@@ -256,6 +256,10 @@ func CountOrganizations() int64 {
 
 // DeleteOrganization completely and permanently deletes everything of organization.
 func DeleteOrganization(org *User) (err error) {
+	if !org.IsOrganization() {
+		return fmt.Errorf("%s is an user not a organization", org.Name)
```
Comment (bagasme) [inline models/org.go]:
```suggestion
		return fmt.Errorf("%s is an user not an organization", org.Name)
```

`a`, `an` infinitive

Comment (zeripath) [reply models/org.go]:
```suggestion
		return fmt.Errorf("%s is a user not an organization", org.Name)
```

**Review by lafriks — APPROVED** [review]

**Comment by 6543** [comment]
ready to merge :rocket:

**Pull request merged**